lib/rubocop/cop/style/trailing_underscore_variable.rb

```diff
@@ -89,35 +91,51 @@ def allow_named_underscore_variables
             cop_config['AllowNamedUnderscoreVariables']
         end
 
-        def unneeded_range(node)
-          left, right = *node
-          variables = *left
+        def unneeded_ranges(node)
```

Comment:
Perhaps this method could return a compacted array, so we don't need to do `nil` checks every time the returned array is iterated over?

Reply:
I would prefer it this way, to be honest - I would need to modify both lines 98 and 122 to include the checks. Line 122, in particular, requires me to either use `.compact` or use `inject`; I think it would be easier this way :)

Reply:
It seems that `#children_offenses` is the only source of `nil`s, due to the conditional inside `#flat_map`, but you could also replace this with `#select`:

```ruby
variables.select(&:mlhs_type?).flat_map { |v| unneeded_ranges(v) }
```

Reply:
Wow, first time hearing that Ruby idiom!